makefiles/info.inc.mk

```diff
@@ -114,6 +114,70 @@ info-build:
 	@echo ''
 	@echo -e 'MAKEFILE_LIST:$(patsubst %, \n\t%, $(abspath $(MAKEFILE_LIST)))'
 
+define json_string_or_null
+$(or $(1:%="%"), null)
+endef
+
+space :=
+space +=
+comma := ,
+
+# Convert a space separated list to a JSON array
+define _to_json_string_list
+[$(filter-out "","$(subst $(space),"$(comma)$(space)",$(1))")]
+endef
+
+# Strips out any existing quotes so that the generated JSON is valid, not necessary sensible
```

Comment:
Any ideas where this could be an issue? I'm thinking in cases where you have to do some escaping magic as a corner case:

```
/* DOCKER_ENVIRONMENT_CMDLINE=$'-e CFLAGS=-DSTRING_FROM_DOCKER=\'\\\"with\ space\\\"\''*/
```

But maybe this is not a problem, anyway a better disclaimer on the actual `recipe` would be usefull!

Reply:
In situation such as here: [`-DNIMBLE_HOST_PRIO="(NIMBLE_CONTROLLER_PRIO + 1)"`](https://github.com/RIOT-OS/RIOT/blob/master/pkg/nimble/Makefile.include#L89). 

> But maybe this is not a problem, anyway a better disclaimer on the actual `recipe` would be usefull!

Ok!

Reply:
I've added a comment block to the recipe explaining the edge case.